### DISCUSSION TITLE
Standalone build

### DISCUSSION THREAD

**Discussion by RemyMachado** [apollographql/apollo-server#7544] [Q&A] [answered]
Is there a way to build a standalone apollo-server than I can upload to my hosting server and just run with node.js without having to import the source code?

I successfully made a standalone of the frontend app with Next.js, but I'm struggling with the backend apollo-server.

Any help would be greatly appreciated.

**Comment by trevor-scheer** [comment]
I don't quite follow the question, what do you mean by
> without having to import the source code

Do you mean that you don't want to ship the Apollo Server source `.ts` files in your deploy to the hosting server? For that you could use a build tool, something like webpack that will bundle your server app.

If that's not what you're asking I'll need a bit more clarity to understand what you're trying to do / what you're struggling to accomplish.

**Reply by RemyMachado** [reply]
I'll try to be more clear.
I would like to avoid cloning the whole project source code on the hosting machine. I would prefer to build the backend on my machine before uploading it to the hosting machine and then run the `dist` folder. Kind of like using an `.exe`. For instance with Next.js you can build a `standalone` version and run it just with the built folder, without needing the package.json nor the original source code.

I hope it makes sense. It's probably doable with Webpack, but I have literally no idea on how to do that so I was wondering if there was a command line or a workaround already available with apollo-server.

**Reply by trevor-scheer** [reply, marked as answer]
Can you build your Apollo Server endpoint into your next app and get the desired outcome? There's an existing community integration for running Apollo Server in a Next app.

I'm almost certain you're asking for a bundler. We don't offer a ready to go solution for that, but it shouldn't be too hard to get one running against your project in short time. This is most common for serverless applications where bundle size is important.

**Reply by RemyMachado** [reply]
Unfortunately, I can't build my apollo-server into my next app. Yes, a bundler. You answered my question:
> a bundler. We don't offer a ready to go solution for that

Thank you.
